components/camel-grok/src/main/java/org/apache/camel/component/grok/GrokDataFormat.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.grok;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.nio.CharBuffer;
+import java.util.*;

```

Comment:
There is a checkstyle issue here I think

Reply:
Strange, Maven plugin doesnt see this.

> [INFO] --- maven-checkstyle-plugin:3.0.0:checkstyle (default-cli) @ camel-grok ---
[INFO] Starting audit...
Audit done.

I expect you mean that I should expand that java.util.* import, right?

Reply:
java.util looks excluded from AvoidStarImport module - https://github.com/apache/camel/blob/master/buildingtools/src/main/resources/camel-checkstyle.xml

Reply:
Ah ok